### PR TITLE
Add Yarn script command to type-check only (small version)

### PR DESCRIPTION
**Pull request by lazytype**
Alternative implementation to #450. The primary downside is that VSCode still won't show the correct type errors in test files.

## ✍️ Proposed changes

🎟 _Jira ticket:_ [Add a dedicated TS check script to package.json](https://jira.mongodb.org/browse/PD-603)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- ~[ ] I have added tests that prove my fix is effective or that my feature works~
- ~[ ] I have added necessary documentation (if appropriate)~
- ~[ ] I have run `yarn changeset` and documented my changes~

- ~[ ] I have added my new package to the global tsconfig~
- ~[ ] I have added my new package to the Table of Contents on the global README~

### CODE REVIEW

**Comment by lazytype** [comment]
> First, it looks like the table package doesn't love the `yarn ts` command, though I think the issues aren't related to the changes here. Still, we should probably fix those errors here so that the command passes.

Yep, I have a separate PR to address the Table issues here https://github.com/mongodb/leafygreen-ui/pull/556

File: tsconfig.json [hunk 1]
```diff
@@ -1,111 +1,19 @@
...
-    },
-    {
-      "path": "./packages/leafygreen-provider"
-    },
-    {
-      "path": "./packages/typography"
-    }
-  ]
+  "include": ["packages/**/src/**/*", "typings/**/*"],
+  "compilerOptions": {
```
Comment (hswolff) [inline tsconfig.json:3]: Anyway we can extend from package.tsconfig.json as the settings should be mostly the same? Or vice versa?

**Comment by DesignerDave** [comment]
> > First, it looks like the table package doesn't love the `yarn ts` command, though I think the issues aren't related to the changes here. Still, we should probably fix those errors here so that the command passes.
> 
> Yep, I have a separate PR to address the Table issues here #556

Ahh, I see! Looks good from my perspective then.

File: common.tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
```
Comment (hswolff) [inline common.tsconfig.json:2]: :chefskiss:

**Pull request merged**